validator/validation/batches.py

```diff
@@ -118,21 +157,108 @@ def reshape_meta(meta):
+) -> dict:
+    """
+    Create a lookup table that aggregates the non-reference measurement points falling under the same reference
+    pixel
+
+    Parameters
+    ----------
+    validation_run: models.validation_run.ValidationRun object
+        Configuration of the run
+    datasets : dict of dicts
+        :Keys: string, datasets names
+        :Values: dict, containing the following fields (see pytesmo DataManager for details):
+            *'class'
+            *'columns'
+            *'args': list, optional
+            *'kwargs': dict, optional
+            *'grids_compatible': boolean, optional
+            *'use_lut': boolean, optional
+            *'lut_max_dist': float, optional
+    ref_name: str
+        Name of the reference dataset
+
+    Returns
+    -------
+    lut: dict
+        lookup table with shape {'other_dataset':{ref gpi: [other gpis]}}
+    """
+    ref_reader = datasets[ref_name]["class"]
+    # get grid of reference
+    while hasattr(ref_reader, 'cls'):
+        ref_reader = ref_reader.cls
+    ref_grid = ref_reader.grid
+
+    lut = {}
+    # a bit of a kack to match dataset name and configuation
+    for other_name, other_config in zip(
+        datasets.keys(),
+        validation_run.dataset_configurations.all(),
+    ):
+        if other_name == ref_name:
+            continue
+        else:
+            other_reader = datasets[other_name]["class"]
+            while hasattr(other_reader, 'cls'):
+                other_reader = other_reader.cls
+            # get all 'other' points, divided in 'jobs'
+            other_points_jobs = create_jobs(
+                validation_run=validation_run,
+                reader=other_reader,
+                dataset_config=other_config,
+                return_points=False
+            )
+            if not other_points_jobs:
+                __logger.debug(
+                    "There are no points to average in the selected dataset "
+                    "of {}. Check the filtering configuration".format(other_config.dataset)
+                )
+                # make sure there is always a key, even when no points are found
+                lut[other_name] = []
+            else:
+                # iterate from the side of the non-reference
+                for other_points in other_points_jobs:
+                    gpis, lons, lats = other_points[0], other_points[1], other_points[2]
+                    other_lut = {}
+                    for gpi, lon, lat in zip(gpis, lons, lats):
+                        # list all non-ref points under the same ref gpi
+                        ref_gpi = ref_grid.find_nearest_gpi(lon, lat)[0]  # todo: implement methods here to combine irregular grids
+                        if ref_gpi in other_lut.keys():
+                            other_lut[ref_gpi].append((gpi, lon, lat))
+                        else:
+                            other_lut[ref_gpi] = [(gpi, lon, lat)]
+
+                    lut[other_name] = other_lut
```

Comment:
This is indented too deeply, should be one level higher, right?

Reply:
You are right, it might have led to problems with multiple jobs. BTW, is there a limit of how big the lookup table can be for efficiency? e.g. I imagine in a global validation

Reply:
I don't know, but this way you would have overwritten the old value every iteration.

Reply:
You're right, but I didn't notice because for small datasets there is only 1 job (1 iteration)